HungarianAlgorithm/Hungarian/Hungarian.cs

```diff
@@ -8,35 +9,206 @@ public class Hungarian
 
         private readonly double[,] _distances; // [house_index, well_index]
 
-        private readonly IUndirectedGraph<int, Edge<int>> _graph;
+        private readonly VertexLabelledGraph _graph;
 
         public Hungarian(ProblemInstance problem)
         {
             _problemInstance = problem;
-            _graph = GraphUtils.CreateGraphBasedOnProblemInstance(problem);
             _distances = CreateDistancesMatrixBasedOnProblemInstance(problem);
+            var graph = GraphUtils.CreateGraphBasedOnProblemInstance(problem);
+
+            // TODO: Move creation of LabelledGraph elsewhere
+            int n = _problemInstance.N * _problemInstance.K;
+            var taggedEdges = graph.Edges.Select(e =>
+            {
+                double cost = e.Source < n ? GetDistanceBetweenHouseAndWell(e.Source, e.Target) : GetDistanceBetweenHouseAndWell(e.Target, e.Source);
+                return new TaggedUndirectedEdge<int, double>(e.Source, e.Target, cost);
+            });
+            var taggedEdgesGraph = new UndirectedGraph<int, TaggedUndirectedEdge<int, double>>();
+            taggedEdgesGraph.AddVertexRange(Enumerable.Range(0, 2 * n));
+            taggedEdgesGraph.AddEdgeRange(taggedEdges);
+
+            _graph = new VertexLabelledGraph(taggedEdgesGraph);
         }
 
-        //var solution = new QuikGraph.Algorithms.Assignment.HungarianAlgorithm(costs);
         public Solution Solve()
         {
-            GraphUtils.WriteAllEdgesOfBaseGraphWithDistances(_graph, _distances);
+            int N = _problemInstance.N * _problemInstance.K;
+            var Matching = new EdgeList<int, Edge<int>>();
+            
+            GetStartingPotential(_graph, out double[] _, out IMutableBidirectionalGraph<int, Edge<int>> eqGraph);
+            var H = Enumerable.Range(0, N).ToHashSet();     // all house vertices
+            var W = Enumerable.Range(N, N).ToHashSet();     // all well vertices
```

Comment:
`Enumerable.Range(N, 2*N)`? Narazie nigdzie chyba z tego nie korzystasz, więc non problemos :)

Reply:
Pierwszy argument to `start`, drugi to `count`

Reply:
Aaa sorki. Tak to jest jak się nie czyta dokumentacji